src/back/glsl/mod.rs

```diff
@@ -1487,6 +1492,41 @@ impl<'a, W: Write> Writer<'a, W> {
         Ok(())
     }
 
+    fn write_workgroup_variables_initialization(
+        &mut self,
+        ctx: &back::FunctionCtx,
+    ) -> BackendResult {
+        if !self.options.zero_initialize_workgroup_memory
+            || !ctx.ty.is_compute_entry_point(self.module)
```

Comment:
> comment: Huh, I actually don't know why whether this is a compute shader matters here. 🤔

Variables in the workgroup address space can only be used by compute entry points.

> it might be a bit clearer to pull this condition out and wrap its negation around the call to `write_workgroup_variables_initialization`.

Will do!